### PR TITLE
[ROCm] more 6.0 changes

### PR DESCRIPTION
**Pull request by jeffdaily**
cc @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

### CODE REVIEW

**Review by pruthvistony — APPROVED** [review]

**Review by huydhn — APPROVED** [review]
Stamped to unblock

**Review by malfet — APPROVED** [review]

File: cmake/public/LoadHIP.cmake [hunk 1]
```diff
@@ -188,7 +188,7 @@ if(HIP_FOUND)
     find_library(PYTORCH_RCCL_LIBRARIES ${rccl_LIBRARIES} HINTS ${ROCM_PATH}/lib)
   endif()
   # hiprtc is part of HIP
```
Comment (pruthvistony) [inline cmake/public/LoadHIP.cmake:190]: NIT: Can we remove this line.

File: cmake/public/LoadHIP.cmake [hunk 2]
```diff
@@ -184,7 +184,7 @@ if(HIP_FOUND)
...
```
Comment (jeffdaily) [inline cmake/public/LoadHIP.cmake:186]:
```suggestion
```

**Pull request closed without merging**